test/e2e/mock-server-json-rpc.ts

```diff
@@ -0,0 +1,68 @@
+import { mockJsonRpcResult } from '../mocks/json-rpc-result';
+
+type RequestConfig = [
+  method: string,
+  options?: {
+    /** optional arbitrary method variant name to return various result values */
+    methodResultVariant?: string;
+    /** optional params value for JSON request used in withJsonBodyIncluding() */
+    params?: any;
+    /** optional result value returned in JSON response */
+    result?: any;
+  },
+];
+
+const DEFAULT_VARIANT = 'default';
+
+/**
+ * Helper function to assist with mocking JSON-RPC POST requests
+ *
+ * @param mockServer
+ * @param listOfRequestConfigs
+ * @example
+ * ```
+ *  await mockServerJsonRpc(mockServer, [
+ *    ['eth_call', {
+ *      params: ['0x52cbe0f49ccdd4dc6e9c13bab024eabd2842045b'],
+ *      result: '0x0000000000000000000000000000000000000000000000000ddfe4d79cbd3de5',
+ *    }],
+ *    ['eth_call', {
+ *      methodResultVariant: 'balanceChecker',
+ *      params: [{to :'0x52cbe0f49ccdd4dc6e9c13bab024eabd2842045b'}],
+ *    }],
+ *    ['eth_gasPrice],
+ *    ['eth_getBalance', {
+ *      result: '0x27d17a5b79f77509541',
+ *    }],
+ *    ['eth_getBlockByNumber']
+ *  ]);
+ * ```
+ */
+async function mockServerJsonRpc(
+  mockServer: any, // MockttpServer
```

Comment:
Can this be typed?

Reply:
thx for the nudge @Mrtenz!

updated [46d77c5](https://github.com/MetaMask/metamask-extension/pull/21086/commits/46d77c58dbeafb43d71c2d8a956d7b13328b516e)